utilities/include/vec4.hpp

```diff
@@ -0,0 +1,149 @@
+#ifndef VEC4_HPP
+#define VEC4_HPP
+
+#include <ostream>
+#include <cmath>
+#include <xsimd/xsimd.hpp>
+
+#define ALPHA_ 0 //include alpha or not.
```

Comment:
there's an optional argument provided in the vec4 class which decides whether to keep the fourth channel into some calculations. the default value of this argument is defined by `ALPHA_` . But if we plan to create a `Vec3` inherited from `Vec4`, we can do that, and yeah it will be better to do that.

Reply:
Yeah, the design which will require the least amount of changes later on, is defining `vec3` to be a child class of `vec4`.